# Using Different Models with Autogen

## OpenAI - Already seen

# Ollama - Local hosted

In [3]:
from autogen_core.models import UserMessage
from autogen_ext.models.ollama import OllamaChatCompletionClient

# Assuming your Ollama server is running locally on port 11434.
ollama_model_client = OllamaChatCompletionClient(model="llama3.2")

response = await ollama_model_client.create([UserMessage(content="Who are you, who created you?", source="user")])
print(response)
await ollama_model_client.close()


finish_reason='stop' content='I\'m an artificial intelligence model known as Llama. Llama stands for "Large Language Model Meta AI."' usage=RequestUsage(prompt_tokens=33, completion_tokens=23) cached=False logprobs=None thought=None


finish_reason='stop' content='I\'m an artificial intelligence model known as Llama. Llama stands for "Large Language Model Meta AI."' usage=RequestUsage(prompt_tokens=33, completion_tokens=23) cached=False logprobs=None thought=None

In [ ]:
from autogen_core.models import UserMessage
from autogen_ext.models.ollama import OllamaChatCompletionClient

# Assuming your Ollama server is running locally on port 11434.
ollama_model_client = OllamaChatCompletionClient(model="llama3.2")

response = await ollama_model_client.create([UserMessage(content="Who are you, who created you?", source="user")])
print(response)


finish_reason='stop' content='I am an artificial intelligence model known as Llama. Llama stands for "Large Language Model Meta AI."' usage=RequestUsage(prompt_tokens=33, completion_tokens=23) cached=False logprobs=None thought=None


finish_reason='stop' content='I am an artificial intelligence model known as Llama. Llama stands for "Large Language Model Meta AI."' usage=RequestUsage(prompt_tokens=33, completion_tokens=23) cached=False logprobs=None thought=None

In [6]:
from autogen_agentchat.agents import AssistantAgent
agent_1 = AssistantAgent(name='MyOllamaAgent',model_client=ollama_model_client)

In [7]:
result = await agent_1. run(task='Tell me something about you?')

In [8]:
result

TaskResult(messages=[TextMessage(id='a6b436b1-ca2a-430f-bf42-cb46e2ac0185', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 6, 598002, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='ad293544-9fd3-4610-9826-536bf9855c2c', source='MyOllamaAgent', models_usage=RequestUsage(prompt_tokens=55, completion_tokens=136), metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 16, 631317, tzinfo=datetime.timezone.utc), content="I'm an artificial intelligence designed to assist and communicate with users like you. I have been trained on a vast amount of text data, which enables me to understand and respond to a wide range of questions and topics.\n\nMy primary function is to provide helpful and accurate information, answer questions, and engage in conversations to the best of my abilities. I can process natural language inputs and generate human-like responses.\n\nI don't have personal 

TaskResult(messages=[TextMessage(id='a6b436b1-ca2a-430f-bf42-cb46e2ac0185', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 6, 598002, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='ad293544-9fd3-4610-9826-536bf9855c2c', source='MyOllamaAgent', models_usage=RequestUsage(prompt_tokens=55, completion_tokens=136), metadata={}, created_at=datetime.datetime(2025, 7, 5, 5, 19, 16, 631317, tzinfo=datetime.timezone.utc), content="I'm an artificial intelligence designed to assist and communicate with users like you. I have been trained on a vast amount of text data, which enables me to understand and respond to a wide range of questions and topics.\n\nMy primary function is to provide helpful and accurate information, answer questions, and engage in conversations to the best of my abilities. I can process natural language inputs and generate human-like responses.\n\nI don't have personal experiences, emotions, or opinions like humans do, but I'm designed to be neutral and informative. My goal is to help users find the information they need, learn new things, or simply have a conversation.\n\nWhat would you like to talk about?", type='TextMessage')], stop_reason=None)

# GEMINI

In [1]:
from autogen_core.models import UserMessage
from autogen_ext.models.openai import OpenAIChatCompletionClient

In [2]:
import os
from dotenv import load_dotenv

load_dotenv()

True

In [3]:
api_key= os.getenv("GOOGLE_API_KEY")
open_router_api_key= os.getenv("OPEN_ROUTER_API_KEY")

In [4]:
model_client = OpenAIChatCompletionClient(
    model="gemini-1.5-flash-8b",
    api_key=api_key,
)

response = await model_client.create([UserMessage(content="What is the capital of India?", source="user")])
print(response)
await model_client.close()

finish_reason='stop' content='New Delhi\n' usage=RequestUsage(prompt_tokens=7, completion_tokens=3) cached=False logprobs=None thought=None


finish_reason='stop' content='New Delhi\n' usage=RequestUsage(prompt_tokens=7, completion_tokens=3) cached=False logprobs=None thought=None

# Open Router

In [ ]:
open_router_model_client=  OpenAIChatCompletionClient(
    base_url="https://openrouter.ai/api/v1",
    # model="openrouter/cypher-alpha:free",
    model="deepseek/deepseek-chat-v3-0324:free",
    api_key=open_router_api_key,
    model_info={
        "family":'deepseek',
        "vision" :True,
        "function_calling":True,
        "json_output": False
    }
)


response= await open_router_model_client.create([UserMessage(content="What is the capital of France?", source="user")])

print(response)

finish_reason='stop' content='The capital of France is **Paris**. It is one of the most famous cities in the world, known for its landmarks like the Eiffel Tower, the Louvre Museum, and Notre-Dame Cathedral.  \n\nWould you like recommendations for places to visit in Paris? 😊' usage=RequestUsage(prompt_tokens=10, completion_tokens=56) cached=False logprobs=None thought=None


finish_reason='stop' content='The capital of France is **Paris**. It is one of the most famous cities in the world, known for its landmarks like the Eiffel Tower, the Louvre Museum, and Notre-Dame Cathedral.  \n\nWould you like recommendations for places to visit in Paris? 😊' usage=RequestUsage(prompt_tokens=10, completion_tokens=56) cached=False logprobs=None thought=None

In [ ]:
response= await open_router_model_client.create([UserMessage(content="What is the capital of India?", source="user")])

In [9]:
response

CreateResult(finish_reason='stop', content='The capital of India is **New Delhi**. \n\nNew Delhi is a part of the larger National Capital Territory (NCT) of Delhi and serves as the political and administrative center of the country. It houses key government institutions, including the Parliament of India and the official residences of the President and Prime Minister. \n\nWould you like information on its history, significance, or attractions?', usage=RequestUsage(prompt_tokens=12, completion_tokens=76), cached=False, logprobs=None, thought=None)

CreateResult(finish_reason='stop', content='The capital of India is **New Delhi**. \n\nNew Delhi is a part of the larger National Capital Territory (NCT) of Delhi and serves as the political and administrative center of the country. It houses key government institutions, including the Parliament of India and the official residences of the President and Prime Minister. \n\nWould you like information on its history, significance, or attractions?', usage=RequestUsage(prompt_tokens=12, completion_tokens=76), cached=False, logprobs=None, thought=None)

In [11]:
response= await open_router_model_client.create([UserMessage(content="What is Autogen Framework, explain in breif?", source="user")])

In [12]:
response

CreateResult(finish_reason='stop', content='The **AutoGen Framework** is an open-source development framework designed to simplify the creation, management, and orchestration of **multi-agent AI systems**. Developed by Microsoft, it enables the deployment of customizable and conversable agents that can collaborate to solve complex tasks through automated interactions.\n\n### **Key Features of AutoGen:**\n1. **Multi-Agent Collaboration**:  \n   - Allows multiple AI agents (e.g., LLM-based assistants, coding agents, task-specific bots) to work together.\n   - Agents can autonomously communicate and delegate tasks among themselves.\n\n2. **Conversable & Customizable Agents**:  \n   - Agents can be personalized with different roles (e.g., Assistant, User Proxy, Coder).\n   - Supports human-in-the-loop interactions for supervision.\n\n3. **Seamless LLM Integration**:  \n   - Compatible with various LLMs (GPT-4, Claude, open-source models).\n   - Supports function calling and code execution.

CreateResult(finish_reason='stop', content='The **AutoGen Framework** is an open-source development framework designed to simplify the creation, management, and orchestration of **multi-agent AI systems**. Developed by Microsoft, it enables the deployment of customizable and conversable agents that can collaborate to solve complex tasks through automated interactions.\n\n### **Key Features of AutoGen:**\n1. **Multi-Agent Collaboration**:  \n   - Allows multiple AI agents (e.g., LLM-based assistants, coding agents, task-specific bots) to work together.\n   - Agents can autonomously communicate and delegate tasks among themselves.\n\n2. **Conversable & Customizable Agents**:  \n   - Agents can be personalized with different roles (e.g., Assistant, User Proxy, Coder).\n   - Supports human-in-the-loop interactions for supervision.\n\n3. **Seamless LLM Integration**:  \n   - Compatible with various LLMs (GPT-4, Claude, open-source models).\n   - Supports function calling and code execution.\n\n4. **Automated Problem-Solving**:  \n   - Agents can dynamically coordinate to debug, optimize, and refine solutions.\n   - Useful for coding, data analysis, decision-making, and automation workflows.\n\n5. **Flexible Deployment**:  \n   - Works in local environments, cloud, or hybrid setups.\n   - Supports Python and REST API integrations.\n\n### **Use Cases:**\n- **Automated Code Generation & Debugging**  \n- **Multi-Agent AI Assistants**  \n- **Task Automation & Workflow Optimization**  \n- **Intelligent Data Processing & Analysis**  \n\nAutoGen simplifies the development of multi-agent AI systems, making it useful for developers, researchers, and enterprises.  \n\nWould you like a deeper dive into any specific feature?', usage=RequestUsage(prompt_tokens=16, completion_tokens=348), cached=False, logprobs=None, thought=None)